doc/sphinx/language/core/variants.rst

```diff
@@ -24,6 +43,37 @@ Variants
    .. exn:: The @natural th argument of @ident must be @ident in @type.
       :undocumented:
 
+.. example::
+
+  The Booleans, the unit type and the empty types are respectively defined by:
+
+   .. coqtop:: in
+
+      Variant bool : Type := true : bool | false : bool.
+      Variant unit : Type := tt : unit.
+      Variant Empty_set : Type :=.
+
+  The option and sum types are defined by:
+
+   .. coqtop:: in
+
+      Variant option (A : Type) : Type := None : option A | Some : A -> option A.
+      Variant sum (A B : Type) : Type := inl : A -> sum A B | inr : B -> sum A B.
+
+  *Boolean reflection* is a relation reflecting under the form of a
```

Comment:
FWIW, "reflection" is probably worth adding to the glossary at some point.

Reply:
It is somehow more specfic to the library than to the logic.